include/jlcxx/module.hpp

```diff
@@ -696,10 +750,29 @@ class JLCXX_API Module
   template<typename T, typename SuperParametersT, typename JLSuperT>
   TypeWrapper<T> add_type_internal(const std::string& name, JLSuperT* super);
 
+  template<typename LambdaT>
+  FunctionWrapperBase& add_lambda(const std::string& name, LambdaT&& lambda, detail::ExtraFunctionData&& extraData)
+  {
+    return lambda_helper(name, std::forward<LambdaT>(lambda), &LambdaT::operator(), std::move(extraData));
+  }
+
   template<typename R, typename LambdaT, typename... ArgsT>
-  FunctionWrapperBase& add_lambda(const std::string& name, LambdaT&& lambda, R(LambdaT::*)(ArgsT...) const)
+  FunctionWrapperBase& lambda_helper(const std::string& name, LambdaT&& lambda, R(LambdaT::*)(ArgsT...) const, detail::ExtraFunctionData&& extraData)
   {
-    return method(name, std::function<R(ArgsT...)>(std::forward<LambdaT>(lambda)));
+    return method_helper(name, std::function<R(ArgsT...)>(std::forward<LambdaT>(lambda)), std::move(extraData));
+  }
+
+  template<typename R, typename... Args, typename... Extra>
```

Comment:
Ah that's actually a bug from moving code around while implementing this.

I tried to fix it (let's see if MSVC is happy now).
And I rebased onto the current main branch + force-push to avoid merge-conflicts (no actual changes/conflicts).